pkg/client/vcwallet/client.go

```diff
@@ -9,25 +9,139 @@ package vcwallet
 import (
 	"encoding/json"
 	"fmt"
+	"time"
+
+	"github.com/hyperledger/aries-framework-go/pkg/secretlock"
+	"github.com/hyperledger/aries-framework-go/pkg/storage"
 )
 
 // provider contains dependencies for the verifiable credential wallet client
 // and is typically created by using aries.Context().
 type provider interface {
-	// TODO to be added #2433
+	StorageProvider() storage.Provider
+}
+
+// kmsOpts contains options for creating verifiable credential wallet client.
+type kmsOpts struct {
+	// local kms options
+	secretLockSvc secretlock.Service
+	passphrase    string
```

Comment:
local and remote kms are mutually exclusive, it's important to check if either one of these options is set, not both and not none.

Reply:
yes they are mutually exclusive, I have defined in option comments about the precedence. Which one takes priority when all options are supplied.